packages/kcms/src/match/controller.ts

```diff
@@ -0,0 +1,48 @@
+import { Result } from '@mikuroxina/mini-fn';
+import { MatchType } from 'config';
+import { SnowflakeIDGenerator } from '../id/main';
+import { TeamID } from '../team/models/team';
+import { MainMatchID } from './model/main';
+import { PreMatchID } from './model/pre';
+import { MainMatchRepository, PreMatchRepository } from './model/repository';
+import { CreateRunResultArgs, FinishState } from './model/runResult';
+import { CreateRunResultService } from './service/createRunResult';
+type MatchResults = {
+  teamID: TeamID;
+  points: number;
+  goalTimeSeconds: number;
+  finishState: FinishState;
+}[];
+export class Controller {
+  private readonly createResult: CreateRunResultService;
+  constructor(
+    idGenerator: SnowflakeIDGenerator,
+    preMatchRepository: PreMatchRepository,
+    mainMatchRepository: MainMatchRepository
```

Comment:
こうできます(こう書くとコンストラクタ内での代入をしなくて済みます)
```suggestion
  constructor(
    private readonly createResult: CreateRunResultService,
    private readonly idGenerator: SnowflakeIDGenerator,
    private readonly preMatchRepository: PreMatchRepository,
    private readonly mainMatchRepository: MainMatchRepository
```